marconi-sidechain/src/Marconi/Sidechain/Api/Routes.hs

```diff
@@ -293,17 +294,85 @@ instance FromJSON AssetIdTxResult where
       <*> v .: "txId"
       <*> v .: "redeemerHash"
       <*> v .: "redeemer"
+      <*> v .: "assetName"
       <*> v .: "burnAmount"
   parseJSON _ = mempty
 
-newtype GetEpochStakePoolDelegationResult
-  = GetEpochStakePoolDelegationResult [EpochState.EpochSDDRow]
-  deriving (Eq, Ord, Show, Generic, ToJSON, FromJSON)
-
 newtype GetEpochActiveStakePoolDelegationResult
-  = GetEpochActiveStakePoolDelegationResult [EpochState.EpochSDDRow]
+  = GetEpochActiveStakePoolDelegationResult [ActiveSDDResult]
   deriving (Eq, Ord, Show, Generic, ToJSON, FromJSON)
 
+data ActiveSDDResult
+  = ActiveSDDResult
+      !C.PoolId
+      !C.Lovelace
+      !C.SlotNo
+      !(C.Hash C.BlockHeader)
+      !C.BlockNo
+  deriving (Eq, Ord, Show)
+
+instance FromJSON ActiveSDDResult where
+  parseJSON (Object v) =
+    ActiveSDDResult
+      <$> v .: "poolId"
+      <*> v .: "lovelace"
+      <*> (C.SlotNo <$> v .: "slotNo")
+      <*> v .: "blockHeaderHash"
+      <*> (C.BlockNo <$> v .: "blockNo")
+  parseJSON _ = mempty
+
+instance ToJSON ActiveSDDResult where
+  toJSON
+    ( ActiveSDDResult
+        poolId
+        lovelace
+        (C.SlotNo slotNo)
+        blockHeaderHash
+        (C.BlockNo blockNo)
+      ) =
+      object
+        [ "poolId" .= poolId
+        , "lovelace" .= lovelace
+        , "slotNo" .= slotNo
+        , "blockHeaderHash" .= blockHeaderHash
+        , "blockNo" .= blockNo
+        ]
+
 newtype GetEpochNonceResult
-  = GetEpochNonceResult (Maybe EpochState.EpochNonceRow)
+  = GetEpochNonceResult (Maybe NonceResult)
   deriving (Eq, Ord, Show, Generic, ToJSON, FromJSON)
+
+data NonceResult
+  = NonceResult
+      !Ledger.Nonce
+      !C.SlotNo
+      !(C.Hash C.BlockHeader)
+      !C.BlockNo
+  deriving (Eq, Ord, Show)
+
+instance FromJSON NonceResult where
+  parseJSON (Object v) =
+    NonceResult
+      <$> (Ledger.Nonce <$> v .: "nonce")
+      <*> (C.SlotNo <$> v .: "slotNo")
+      <*> v .: "blockHeaderHash"
+      <*> (C.BlockNo <$> v .: "blockNo")
+  parseJSON _ = mempty
```

Comment:
I've introduced [`withObject`](https://hackage.haskell.org/package/aeson-2.1.2.1/docs/Data-Aeson.html#v:withObject) in the PR that serialise/deserialise `Value`, added it to other JSON deserialisation and I'd suggest that we stick to it to get better error messages.

Reply:
Done